src/Moryx/Serialization/DefaultSerialization.cs

```diff
@@ -73,6 +74,13 @@ public virtual EntryPrototype[] Prototypes(Type memberType, ICustomAttributeProv
         /// <see cref="ICustomSerialization"/>
         public virtual string[] PossibleValues(Type memberType, ICustomAttributeProvider attributeProvider)
         {
+#if NET8_0
+            var validationAttribute = attributeProvider.GetCustomAttribute<AllowedValuesAttribute>();
+            var allowedValues = validationAttribute?.Values.Select(o => o.ToString()) ?? Enumerable.Empty<string>();
+            if (allowedValues.All(value => !string.IsNullOrEmpty(value))) 
```

Comment:
What is the intention behind this if? From my understanding this would cause the allowed values to be used only if the empty string is not part of the AllowedValues, is that intended? 
Also this method will throw a NullRef exception, if someone in a .net 8 application uses the possible values attribute for a list of primitive types or an enum.
Also the PossibleValues attribute is ignored, if the AllowedValues attribute is used (This is the only thing i'm fine with, if you intended it)

This is what i would expect, but please check if i missunderstood something 🙈
```
if( allowedValues is null)
{
    return allowedValues?.Distinct().ToArray();
}
```